sound/core/control.c

```diff
@@ -268,6 +268,8 @@ struct snd_kcontrol *snd_ctl_new1(const struct snd_kcontrol_new *ncontrol,
 	}
 	kctl->id.index = ncontrol->index;
 
+	kctl->kcontrol_type = ncontrol->kcontrol_type;
+
```

Comment:
where is ncontrol->kcontrol_type initialized?

Reply:
in soc-topology.c in soc_tplg_dapm_widget_dmixer_create and related control creation functions.